batch7crud-roll2/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,137 @@
+        employee.setId(resultSet.getInt("id"));
+        employee.setName(resultSet.getString("username"));
+        employee.setAddress(resultSet.getString("address"));
+        employee.setEmail(resultSet.getString("email"));
+        employee.setContact(resultSet.getString("contact"));
+        empList.add(employee);
+      }
+      preStmt.close();
+      con.close();
+      return empList;
+    } catch (SQLException e) {
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public Employee fetchById(int id) throws DataException {
+    try {
+      con = DBConnection.getSqlConnection();
+      Employee employee = null;
+      String qry = "SELECT * FROM employee WHERE id=?";
+      PreparedStatement preStmt = con.prepareStatement(qry);
+      preStmt.setInt(1, id);
+      ResultSet resultSet = preStmt.executeQuery();
+      while (resultSet.next()) {
+        employee = new Employee();
+        employee.setId(resultSet.getInt("id"));
+        employee.setName(resultSet.getString("username"));
+        employee.setAddress(resultSet.getString("address"));
+        employee.setEmail(resultSet.getString("email"));
+        employee.setContact(resultSet.getString("contact"));
+      }
+      preStmt.close();
+      con.close();
+      return employee;
+    } catch (SQLException e) {
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void update(Employee employee) throws DataException {
+    try {
+      con = DBConnection.getSqlConnection();
+      String qry = "UPDATE employee SET username=?, address=?, email=?, contact=? WHERE id=?";
+      PreparedStatement preStmt = con.prepareStatement(qry);
+      preStmt.setString(1, employee.getName());
+      preStmt.setString(2, employee.getAddress());
+      preStmt.setString(3, employee.getEmail());
+      preStmt.setString(4, employee.getContact());
+      preStmt.setInt(5, employee.getId());
+      preStmt.execute();
+      preStmt.close();
+      con.close();
+    } catch (SQLException e) {
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void delete(int empId) throws DataException {
+    try {
+      con = DBConnection.getSqlConnection();
+      String qry = "DELETE FROM employee where id=?";
+      PreparedStatement preStmt = con.prepareStatement(qry);
+      preStmt.setInt(1, empId);
+      preStmt.executeUpdate();
+      preStmt.close();
+      con.close();
+    } catch (SQLException e) {
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public int getTotalNoOfRecords() throws DataException {
+    try {
+      con = DBConnection.getSqlConnection();
+      String qry = "SELECT COUNT(*) AS total FROM employee";
+      PreparedStatement preStmt = con.prepareStatement(qry);
+      ResultSet resultSet = preStmt.executeQuery();
+      int total = 0;
+      while (resultSet.next()) {
+        total = resultSet.getInt("total");
+      }
+      preStmt.close();
+      con.close();
+      return total;
+    } catch (SQLException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)